### PR TITLE
Regenerated translation files

### PR DESCRIPTION
**Pull request by rarila**
- Many fixes to how html code is used
- Use typografically correct apostrophes on ``en_GB``
- Use typografically correct quotation marks on ``en_GB`` contenttypes
- Switch to non nested output for keybased translations for easier mass editing
- Removing of some unused translations
- Added some missing translations

Another regeneration needed needed, but I will add a nut command to do this for all languages with one keypress.

### CODE REVIEW

**Comment by horus68** [comment]
Please change the language folders to xx_XX language code

**Comment by rarila** [comment]
Which ones?

**Comment by bobdenotter** [comment]
As discussed on IRC, I don't think we should make the switch to typographically correct quotation marks in this PR. 

 - I'm not convinced this is the correct approach
 - We're _this_ close to beta, and we're very much in "feature freeze"

Should we decide to go with typographically correct quotation marks, i'd like to discuss this beforehand, to set an approach, and i would _very_ much like to see some data on how other Open Source projects handle this.

**Comment by horus68** [comment]
all languages using 2 letters code should be changed to the 4 letters code.
So: "de_DE" and not "de"
Note: the actual "pt" should be "pt_BR"
I already PR a "pt_PT" for Neutral portuguese (Portugal)

**Comment by horus68** [comment]
Rules for name folders:
> Standard country-language codes are comprised of two arguments separated by a dash, for example fr-CA is French Canadian.
> 
> The first argument is a valid ISO Language Code. These codes are the lower-case two-letter codes as defined by ISO-639. You can find a full list of these codes at a number of sites like here.
> 
> The second argument to both constructors is a valid ISO Country Code. These codes are the upper-case two-letter codes as defined by ISO-3166. You can find a full list of these codes at a number of sites like here

Country Code Language List - http://www.fincher.org/Utilities/CountryLanguageList.shtml

Locales and Language Codes | Bodeux International LLC | Denver, Colorado - http://bodeuxinternational.com/locales-and-lang-codes/

**Comment by rarila** [comment]
Nothing new, but there’s one major reasons why this wasn’t done on these orphaned translations when we did the switch. You simply need someone who knows. As we are all no extremly talented linguists they simply stayed this was.

The ones with two arguments are the ones done/checked by one native speaker, who is able to decide. So for example I converted ``de`` to ``de-DE`` and I’m quite sure that it's not ``de-AT``. Afaik ``pt`` was done by persons from PT and BR, so it probably was even mixed. Don’t ask me :-)

My Spanisch for example starts and ends with ‘Hola!’. Hence I’m not the one who is able to decide if ``es`` should be renamed to ``es-AR``, ``es-VE``, ``es-BO``, ``es-CL``, ``es-CO``, ``es-CR``, ``es-DO``, ``es-EC``, ``es-SV``, ``es-GT``, ``es-HN``, ``es-MX``, ``es-NI``, ``es-PA``, ``es-PY``, ``es-PE``, ``es-PR``, ``es-ES``, ``es-US`` or ``es-UY``. I don't even know without looking up what ``nb`` or ``id`` is. :smile: 

So suggestions welcome.

*I’m glad we don’t go into even further detail like ``az-Arab-IR``, ``zh-Hant-CN`` or ``sr-Latn-RS``* :sweat_smile:

**Comment by horus68** [comment]
As a proof of concept I also included in crowdin this PR as a branch
https://crowdin.com/project/bolt
So you can download translations for actual master or for this PR.
Hope that the 
> Added some missing translations
will also be on master!

**Comment by horus68** [comment]
I'm more used to those codes... 
el = el_GR
es = es_ES
fi = fi_FI
fr = fr_FR
hu = hu_HU
id = id_ID
it = it_IT
ja = ja_JP
nb = nb_NO
pt = pt_BR
ru = ru_RU

as for the actual "pt" i can assure you that the actual translation is pt_BR (and its a good one)

**Comment by GawainLynch** [comment]
> es = es_ES

Be very careful with this one… it will raise as much ire as pt_PT and pt_BR

> as for the actual "pt" i can assure you that the actual translation is pt_BR (and its a good one)

Yeah, you have our trust on that one. :+1: 

Our contributors to the Portuguese translations to this point have all been Brazilian, so it is good to have a native speaker on board :smile:

**Comment by rarila** [comment]
No one said that we shouldn’t change if you can assure this, quite the contrary!

**Comment by rarila** [comment]
@horus68  So, if there are no technically reasons against it I would say we stay with the two letters until one like you with ``pt_PT`` takes over and trims translation ``xx`` into ``xx_XX``. Then let’s happily change.

**Comment by bobdenotter** [comment]
> > es = es_ES
> Be very careful with this one… it will raise as much ire as pt_PT and pt_BR

Perhaps for the ones we're not sure, we could just duplicate them? `es` -> `es_ES` and `es` -> `es_MX`. If then somebody comes along saying our es_MX sucks, they can help fix it.

**Comment by rarila** [comment]
@horus68 … and what the hill is ``English (upside down)`` ?

**Comment by rarila** [comment]
@bobdenotter you saw the list of how many variants are possible for even just ``es``?

**Comment by bobdenotter** [comment]
@rarila ˙`S∩∀‾uǝ` oʇ pǝʇɐlǝɹ ʎlǝsolɔ s,ʇI

**Comment by bobdenotter** [comment]
> you saw the list of how many variants are possible for even just es?

Yes, but there are also way less 'usual suspects'.. Nobody assumes software comes with `nl_FRY`. the people in Frisia just use regular dutch.

**Comment by horus68** [comment]
> and what the hill is English (upside down) 

This is just a control (dummy) language: 
No translators, it gets all contents from the en_GB master files so I can test if Crowdin downloaded files causes issues on Bolt.
You know sometimes translators break things on their languages and claim the system is wrong. This way I know for sure if the system is working fine!

**Comment by horus68** [comment]
> So, if there are no technically reasons against it I would say we stay with the two letters until one like you with pt_PT takes over and trims translation xx into xx_XX. Then let’s happily change.

@rarila 
Here you have a new PR to include pt-PT in master v3: https://github.com/bolt/bolt/pull/5032

**Comment by rarila** [comment]
**Discussion: typographically correct writing in translations**

- No breakage to expected. Things are just simply put into HTML. The German translation has correct quotation marks since about version 1.6 and still nothing exploded.
- 7-bit ASCII is so 1990’s as old typewriters are. Unicode is now around quite a while, so what.
- That is not a feature, just a string to display
- Correct typography makes things look much more professional (Wasn’t that the reason why the poor cute animals had to go?)
- Others might just be lazy or don’t care.
- We should yet take the chance as probably most translations seems to get some updates and put on a guideline that we like correct typography. Nobody want’s to make the work twice.
- Doing such stuff with an algorithm is at least error prune when not impossible as handling grammar.
- Decomposition is always easy, the other way round not.
- That affects not only quotation marks, also apostrophes, dashes, nonbreaking spaces, number formatting, …
- It’s just language dependant data and should follow the rules of that language.
- We also use ``ö``, ``и`` and ``模``. We use ``…`` instead of ``...`` as an ellipse is an elipse and not just three points. So why not ``“`` and ``”``, if that's the correct way to do.

And now the cons:
- :balloon: 
-  :scream:

**Comment by rarila** [comment]
>  ˙S∩∀‾uǝ oʇ pǝʇɐlǝɹ ʎlǝsolɔ s,ʇI

@bobdenotter That was what I thought after reading half of this sentence … ;-)

**Comment by GawainLynch** [comment]
> And now the cons:
>
> * :balloon:
> * :scream:

Mate, you're doing that with the "pros" section.

Seriously… I am putting a #BPFL foot down. 
* We are days away from a year long development cycle going beta (with a 150,000 line delta to 2.2.0)
* The community is growing rapidly and that is putting a lot of stress on the core team
* The core team are all **really tired & grumpy**

Ergo… Now is *not* the time to be making changes that people do not agree with, and have not been discussed and agreed upon… 

"A camel is a horse designed by committee", I'll grant you, but we've all managed to find way to work together and agree on stuff over the last few years… can we please get back to that mode of operation.

@rarila This PR is important and I am "this" close to pressing the merge button… it is needed, and is a **good change** :grin: 

Can you first please revert the typography changes in this PR for English, to ensure tests are passing, and RFC a discussion on typography (copy pasting your above points is fine) and we'll do "whatever" for 3.1.0.

**Comment by CarsonF** [comment]
Wait why is non-nested better??? I definitely don't like that.

**Comment by GawainLynch** [comment]
> Wait why is non-nested better??? I definitely don't like that.

Easier to search for stings, etc, and I believe integration with online services.

**Comment by CarsonF** [comment]
Transifex handles nesting fine

File: app/resources/translations/en_GB/messages.en_GB.yml [hunk 1]
```diff
@@ -106,7 +106,7 @@
 "Files could not be uploaded.": #
 "Generic failure while testing connection to extension server: %errormessage%": #
 "Has to be a floating point number!": #
-"Id (№)": #
+"Id (?)": #
```
Comment (bobdenotter) [inline app/resources/translations/en_GB/messages.en_GB.yml]: My precious №!

File: app/resources/translations/en_GB/messages.en_GB.yml [hunk 2]
```diff
@@ -185,7 +185,7 @@
 
 #--- 297 translations ---------------------------------------------------------
 
-"%name% № %id%": "%name% № %id%"
+"%name% ? %id%": "%name% ? %id%"
```
Comment (bobdenotter) [inline app/resources/translations/en_GB/messages.en_GB.yml]: And here!

File: app/resources/translations/en_GB/messages.en_GB.yml (hunk 1 above)
Comment (GawainLynch) [inline app/resources/translations/en_GB/messages.en_GB.yml]: https://www.youtube.com/watch?v=WWaLxFIVX1s

File: app/resources/translations/en_GB/messages.en_GB.yml (hunk 2 above)
Comment (GawainLynch) [inline app/resources/translations/en_GB/messages.en_GB.yml]: The 5 minute version https://www.youtube.com/watch?v=nF_lJTGKFP0

File: app/resources/translations/en_GB/messages.en_GB.yml (hunk 1 above)
Comment (rarila) [inline app/resources/translations/en_GB/messages.en_GB.yml]: Typographically has no place in this file

**Pull request merged**